### DISCUSSION TITLE
Just to clarify, does this mean disk space can be reclaimed unlike lmdb / boltdb?

### DISCUSSION THREAD

**Discussion by hiqsociety** [erthink/libmdbx#346] [Q&A] [answered]
Does it mean the size can be reclaimed?

**4. Automatic continuous zero-overhead database compactification.**
During each commit libmdbx merges suitable freeing pages into unallocated area at the end of file, and then truncates unused space when a lot enough of.

**Comment by hiqsociety** [comment]
is the min compaction threshold value is the minimum libmdbx will compact?

**Comment by erthink** [comment, marked as answer]
1. See [`mdbx_env_set_geometry()`](https://libmdbx.dqdkfa.ru/group__c__settings.html#ga79065e4f3c5fb2ad37a52b59224d583e), the `size_lower` parameter.
2. Internally, there is no such threshold, since automatic compactification is always useful.
I.e autocompaction is always performed (unless `MDBX_ENABLE_REFUND=0` is defined during build `libmdbx`), but the database file will not be truncated below the `size_lower` value.
